website/docs/guides/container_service_extension.html.markdown

```diff
@@ -0,0 +1,445 @@
+
+  edge_gateway_id = vcd_nsxt_edgegateway.cse_egw.id
+
+  gateway       = "192.168.7.0"
+  prefix_length = 24
+
+  static_ip_pool {
+    start_address = "192.168.7.1"
+    end_address   = "192.168.7.100"
+  }
+
+  dns1 = "8.8.8.8"
+  dns2 = "8.8.8.4"
+}
+```
+
+To be able to reach the Kubernetes nodes within the routed network, you need also a
+[SNAT rule](/providers/vmware/vcd/latest/docs/resources/nsxt_nat_rule):
+
+```hcl
+resource "vcd_nsxt_nat_rule" "snat" {
+  org             = vcd_org.cse_org.name
+  edge_gateway_id = vcd_nsxt_edgegateway.cse_egw.id
+
+  name        = "SNAT rule"
+  rule_type   = "SNAT"
+  description = "description"
+
+  external_address = "88.88.88.89"    # A public IP from the external network
+  internal_address = "192.168.7.0/24" # This is the routed network CIDR
+  logging          = true
+}
+```
+
+### Step 3: Configure ALB
+
+Advanced Load Balancers are required for CSE to be able to handle Kubernetes services and other internal capabilities.
+You need the following resources:
+
+* [ALB Controller](/providers/vmware/vcd/latest/docs/resources/nsxt_alb_controller)
+* [ALB Cloud](/providers/vmware/vcd/latest/docs/resources/nsxt_alb_cloud)
+* [ALB Service Engine Group](/providers/vmware/vcd/latest/docs/resources/nsxt_alb_service_engine_group)
+* [ALB Settings](/providers/vmware/vcd/latest/docs/resources/nsxt_alb_settings)
+* [ALB Edge Gateway Service Engine Group](/providers/vmware/vcd/latest/docs/resources/nsxt_alb_edgegateway_service_engine_group)
+* [ALB Pool](/providers/vmware/vcd/latest/docs/resources/nsxt_alb_pool)
+* [ALB Virtual Service](/providers/vmware/vcd/latest/docs/resources/nsxt_alb_virtual_service)
+
+You can have a look at [this guide](/providers/vmware/vcd/latest/docs/guides/nsxt_alb) as it explains every resource
+and provides some examples of how to set up ALB in VCD. You can also have a look at the "[Examples](#examples)" section below
+where a full ALB setup is provided.
+
+### Step 4: Configure catalogs and OVAs
+
+You need to have a [Catalog](/providers/vmware/vcd/latest/docs/resources/catalog) for vApp Templates and upload the corresponding
+TKGm (Tanzu Kubernetes Grid) OVA files to be able to create Kubernetes clusters.
+
+```hcl
+data "vcd_storage_profile" "cse_storage_profile" {
+  org        = vcd_org.cse_org.name
+  vdc        = vcd_org_vdc.cse_vdc.name
+  name       = "*"
+  depends_on = [vcd_org.cse_org, vcd_org_vdc.cse_vdc]
+}
+
+resource "vcd_catalog" "cat-cse" {
+  org         = vcd_org.cse_org.name
+  name        = "cat-cse"
+  description = "CSE catalog"
+
+  storage_profile_id = data.vcd_storage_profile.cse_storage_profile.id
+
+  delete_force     = "true"
+  delete_recursive = "true"
+  depends_on       = [vcd_org_vdc.cse_vdc]
+}
+```
+
+Then you can upload TKGm OVAs to this catalog. These can be downloaded from **VMware Customer Connect**.
+To upload them, use the [Catalog Item](/providers/vmware/vcd/latest/docs/resources/catalog_item) resource.
+
```

Comment:
Just a note, the OVA for TKGm clusters must be uploaded through the CSE server. This is applicable from CSE v3.1.2 onwards

> cse template install or
> cse template import

Or, if there is a way to expose the metadata to the CSE server if the OVA is uploaded via VCD directly.

Reply:
Yeah, below the example HCL code you can find a paragraph that tells about the required metadata for the CSE server to fetch the uploaded OVAs correctly.

Reply:
I've refined that paragraph a bit.